cron/monthly/expiring-cards.js

```diff
@@ -20,13 +19,15 @@ if (config.env === 'production' && date !== 7 && date !== 21 && !process.env.OFF
 
 // link payment method id in Orders to payment method id in the payment method we're updating
 
-const fetchExpiringCreditCards = async (month, year) => {
+const fetchExpiringCreditCards = async () => {
   const expiringCards = await models.PaymentMethod.findAll({
     where: {
       type: 'creditcard',
-      data: {
-        expMonth: month,
-        expYear: year,
+      // Expiry Date is set to be the last second of the valid month.
+      // This query will return all the creditcards that will expire by the end of the current month.
+      expiryDate: {
+        [Op.gt]: moment.utc().startOf('month'),
+        [Op.lt]: moment.utc().endOf('month'),
```

Comment:
What are the precise values being used in the comparaison practically?

Could it be that we end up with a condition like expiryDate < `2022-01-31 23:59:59` and expiryDate=2022-01-31 23:59:59. Not returning entries.

Reply:
Moment will return the timestamp to the last millisecond, but I'll update to `gte` and `lte` to be more explicit.